createAnimatableComponent.js

```diff
@@ -412,9 +412,21 @@ export default function createAnimatableComponent(WrappedComponent) {
         delay: iterationDelay || 0,
       };
 
-      Animated.timing(animationValue, config).start(endState => {
+      let animation = Animated.timing(animationValue, config);
+      const canLoop =
+        !iterationDelay &&
+        (iterationCount === 'infinite' || iterationCount > 1);
+      if (canLoop) {
+        const loopConfig =
+          iterationCount !== 'infinite' ? { iterations: iterationCount } : {};
+
+        animation = Animated.loop(animation, loopConfig);
+      }
+
+      animation.start(endState => {
         currentIteration += 1;
```

Comment:
I'd prefer to rewrite this completely to only use `Animated.loop` to make it less complicated. For infinite loops you can pass `-1` as iterations.

Reply:
Correct me if I'm wrong, but I don't think you can support `iterationDelay` using `Animated.loop`.

I'd be happy to adjust the code so that single iterations also utilize `Animated.loop`, and that infinite loops use `-1`, but I don't think we can entirely eliminate this conditional.

Reply:
Hmm, wonder if we can solve this with `Animated.delay()` instead, but probably not.

Reply:
I took a more careful look and it appears that simply applying `iterationDelay` within the `Animated.timing` works fine in the context of an `Animated.loop`.

I've updated the PR and I think you'll find it much more succinct (and thus to your liking).